src/IcalParser.php

```diff
@@ -562,7 +562,8 @@ public function getEvents($includeRecurring=false) {
 		$events = [];
 		if(isset($this->data['VEVENT'])) {
 			if($includeRecurring === true) {
-				for ($i = 0; $i < count($this->data['VEVENT']); $i++) {
+                $eventCount = count($this->data['VEVENT']);
+				for ($i = 0; $i < $eventCount; $i++) {
```

Comment:
Small change, big impact. Originally this `for` loop contained: `count($this->data['VEVENT'])`. This means that during each iteration of the loop, php checks to see if index of the current iteration, `$i`, is less than the count/length of the array `$this->data['VEVENT']`, which requires php to _calculate_ that length _every iteration_. So that means that in addition to looping through every single item in the array, during each of those iterations, php is iterating over every element AGAIN to get the count. 

That's O^N for time, where N=number_of_events_in_calendar (even though the core issue in this bug is _memory_, it still takes a really long time to upload calendars, so this does matter). For the ical link related to this ticket, N is somewhere between 10,000 and 11,000. Storing the count in `$eventCount` reduces that to O(1) to get the count, and then O(1) for the actual for loop itself, so O(2)--much better.